src/scanner.cc

```diff
@@ -657,7 +657,7 @@ struct Scanner {
 
     int16_t *ind_ptr = &ind_len_stk.back();
     int16_t cur_ind = *ind_ptr--;
-    int16_t prt_ind = *ind_ptr;
+    int16_t prt_ind = ind_len_stk.size() > 1 ? *ind_ptr : 0;
```

Comment:
Since the `prt_ind` value won't be used if `cur_ind_typ` is `IND_ROT`, which is the only case that unowned memory will be accessed, the `prt_ind` placeholder value (i.e., `0`) does not matter here but I think it'd be better to set it as `-1` so that it'd be more explicit that this value can be anything. And I think using iterators to achieve our goal may be better since it looks more readable to me, what do you think?

For example:

```cpp
vector<int16_t>::reverse_iterator ind_ptr = ind_len_stk.rbegin();
vector<int16_t>::reverse_iterator ind_end = ind_len_stk.rend();
int16_t cur_ind = *ind_ptr++;
int16_t prt_ind = ind_ptr == ind_end ? -1 : *ind_ptr;
```

Reply:
Looks good to me, I'll make those changes